roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/Car.java

```
@@ -59,15 +59,22 @@ public void setLastUpdateTime(double lastUpdateTime)
     }
 
     @Override
-    public double getTimeToTraverseSection() {
-        //TODO getTimeToTraverseSection()
-        return 0;
+    public double getTimeToTraverseCurrentSection() {
+
+        return getTimeToTraverseSection(this.getCurrentSection());
     }
 
     @Override
     public double getTimeToTraverseSection(IStreetSection section) {
-        //TODO getTimeToTraverseSection(IStreetSection section)
-        return 0;
+
+        double carPosition = 0;
+
+        if (section.getCarPositions().containsKey(this)) {
+            carPosition = section.getCarPositions().get(this);
+        }
+
+        double remainingLength = section.getLength() - carPosition;
+        return remainingLength / this.getDriverBehaviour().getSpeed();
```

if car is not in this section I would throw a exception